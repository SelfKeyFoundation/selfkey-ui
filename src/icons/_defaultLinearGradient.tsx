import * as React from 'react';

export const DefaultLinearGradient = () => {
    return (
        <defs>
            <linearGradient x1="11.0842631%" y1="68.33943%" x2="36.0705554%" y2="0%" id="dlg">
                <stop stopColor="#09A8BA" offset="0%"></stop>
                <stop stopColor="#0ABBD0" offset="100%"></stop>
            </linearGradient>
        </defs>
    )
};

export default DefaultLinearGradient;
