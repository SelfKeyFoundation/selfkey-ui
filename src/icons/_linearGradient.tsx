import * as React from 'react';

export type LinearGradientProps = any;

export const LinearGradient = ({ color }: LinearGradientProps) => {
	let color1;
	let color2;
	let id;
	switch (color) {
		case 'orange':
			color1 = '#B44000';
			color2 = '#DB7400';
			id = 'orange-linear-gradient';
			break;
		case 'grey':
			color1 = '#93B0C1';
			color2 = '#647A86';
			id = 'grey-linear-gradient';
			break;
		default:
			color1 = '#09A8BA';
			color2 = '#0ABBD0';
			id = 'default-linear-gradient';
	}

	return (
		<defs>
			<linearGradient x1="11.0842631%" y1="68.33943%" x2="36.0705554%" y2="0%" id={id}>
				<stop stopColor={color1} offset="0%"></stop>
				<stop stopColor={color2} offset="100%"></stop>
			</linearGradient>
		</defs>
	);
};

export default LinearGradient;
