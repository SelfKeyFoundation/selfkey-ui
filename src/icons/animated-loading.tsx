import * as React from 'react';
import { WithStyles, withStyles, createStyles } from '@material-ui/core';
import { SvgIcon } from '@material-ui/core';
import { SvgIconProps } from '@material-ui/core/SvgIcon';

const styles = createStyles({
    root: {
       width: '120px !important',
       height: '30px !important',
    },
});

export const AnimatedLoadingIcon = withStyles(styles)((props: SvgIconProps & WithStyles<typeof styles>) => (
    <SvgIcon {...props} viewBox="0 0 120 30">
        <circle cx="15" cy="15" r="15">
            <animate
                attributeName="r"
                from="15"
                to="15"
                begin="0s"
                dur="0.8s"
                values="15;9;15"
                calcMode="linear"
                repeatCount="indefinite"
            />
            <animate
                attributeName="fillOpacity"
                from="1"
                to="1"
                begin="0s"
                dur="0.8s"
                values="1;.5;1"
                calcMode="linear"
                repeatCount="indefinite"
            />
        </circle>
        <circle cx="60" cy="15" r="9" fillOpacity="0.3">
            <animate
                attributeName="r"
                from="9"
                to="9"
                begin="0s"
                dur="0.8s"
                values="9;15;9"
                calcMode="linear"
                repeatCount="indefinite"
            />
            <animate
                attributeName="fillOpacity"
                from="0.5"
                to="0.5"
                begin="0s"
                dur="0.8s"
                values=".5;1;.5"
                calcMode="linear"
                repeatCount="indefinite"
            />
        </circle>
        <circle cx="105" cy="15" r="15">
            <animate
                attributeName="r"
                from="15"
                to="15"
                begin="0s"
                dur="0.8s"
                values="15;9;15"
                calcMode="linear"
                repeatCount="indefinite"
            />
            <animate
                attributeName="fillOpacity"
                from="1"
                to="1"
                begin="0s"
                dur="0.8s"
                values="1;.5;1"
                calcMode="linear"
                repeatCount="indefinite"
            />
        </circle>
    </SvgIcon>
));

export default AnimatedLoadingIcon;
