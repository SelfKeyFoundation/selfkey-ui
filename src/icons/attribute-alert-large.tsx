import * as React from 'react';
import { SvgIcon, WithStyles, withStyles, createStyles } from '@material-ui/core';
import { SvgIconProps } from '@material-ui/core/SvgIcon';

const styles = createStyles({
    root: {
       width: '38px !important',
       height: '44px !important',
    },
});

export const AttributeAlertLargeIcon = withStyles(styles)((props: SvgIconProps & WithStyles<typeof styles>) => (
	<SvgIcon {...props} viewBox="0 0 38 44" version="1.1">
        <g id="ðŸ’»-ID-Dashboard-attribute-large-icon" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
            <g id="410.1-Marketplace-Applications-Statuses" transform="translate(-195.000000, -379.000000)" fill="#E98548">
                <g id="Content" transform="translate(150.000000, 264.000000)">
                    <g id="Collapse-box-1">
                        <g id="Status-box-1" transform="translate(15.000000, 90.000000)">
                            <path d="M67.91,31.27 C67.9698291,31.5087207 68.0000562,31.7538962 68,32 L68,45.3 C67.9841317,53.5919031 63.5062321,61.23337 56.28,65.3 L50.47,68.57 C49.5571638,69.0831103 48.4428362,69.0831103 47.53,68.57 L41.72,65.3 C34.4937679,61.23337 30.0158683,53.5919031 30,45.3 L30,32 C30,31.32 30.27,29.02 33.66,27.32 C36.73,25.78 41.89,25 49,25 C64.09,25 67.32,28.93 67.91,31.27 Z M65,45.27 L65,32 C65,32 64,28 49,28 C33,28 33,32 33,32 L33,45.3 C32.9995752,52.5239266 36.8947728,59.186664 43.19,62.73 L49,66 L54.81,62.7 C61.1052272,59.156664 65.0004248,52.4939266 65,45.27 Z M48.5,59 C47.6715729,59 47,58.3284271 47,57.5 C47,56.6715729 47.6715729,56 48.5,56 C49.3284271,56 50,56.6715729 50,57.5 C50,58.3284271 49.3284271,59 48.5,59 Z M48.5,33.5 C49.3284271,33.5 50,34.1715729 50,35 L50,51 C50,51.8284271 49.3284271,52.5 48.5,52.5 C47.6715729,52.5 47,51.8284271 47,51 L47,35 C47,34.1715729 47.6715729,33.5 48.5,33.5 Z" id="attribute-large-icon"></path>
                        </g>
                    </g>
                </g>
            </g>
        </g>
	</SvgIcon>
));

export default AttributeAlertLargeIcon;
