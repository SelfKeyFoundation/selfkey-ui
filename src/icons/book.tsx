import * as React from 'react';
import injectSheet, { StyleSheet, StyledComponentProps } from 'react-jss';

import { SvgIcon } from '@material-ui/core';
import { SvgIconProps } from '@material-ui/core/SvgIcon';
import DefaultLinearGradient from './_defaultLinearGradient';

const styles: StyleSheet = {
  root: {
    transition: 'all 0.2s ease-out',
    '&:hover': {
        fill: 'white',
    },
  }
};

export const BookIcon = injectSheet(styles)<SvgIconProps>((props: SvgIconProps & StyledComponentProps)=> (
  <SvgIcon style={{ width: props.width || '68px', height: props.height || '55px', fill: props.fill || 'url(#dlg)' }} {...props} viewBox='0 0 68 55'>
    <DefaultLinearGradient />
    <g id="ðŸ’»-ID-Dashboard" stroke="none" strokeWidth="1" fill="inherit" fillRule="evenodd">
        <g id="403---Identity-Summary" transform="translate(-240.000000, -958.000000)" fill="inherit" fillRule="nonzero">
            <g id="Area-box-2" transform="translate(150.000000, 462.000000)">
                <g id="Additional" transform="translate(14.000000, 390.000000)">
                    <g id="Explanatory" transform="translate(10.000000, 106.000000)">
                        <path d="M71,0 L71,3 L66,3 L66,52 L95.505859,52 C96.284666,53.745175 97.971465,55 100,55 C102.028851,55 103.715042,53.74513 104.494141,52 L134,52 L134,3 L129,3 L129,0 L103,0 C101.789062,0 100.735556,0.576246 100,1.435547 C99.264444,0.576246 98.210938,0 97,0 L71,0 Z M73,2 L97,2 C98.116666,2 99,2.883334 99,4 L101,4 C101,2.883334 101.883334,2 103,2 L127,2 L127,45 L103,45 C101.789062,45 100.735556,45.576246 100,46.435547 C99.264444,45.576246 98.210938,45 97,45 L73,45 L73,2 Z M68,5 L71,5 L71,47 L97,47 C98.116666,47 99,47.883334 99,49 L101,49 C101,47.883334 101.883334,47 103,47 L129,47 L129,5 L132,5 L132,50 L103.085938,50 L102.890625,50.748047 C102.554647,52.042353 101.404427,53 100,53 C98.595573,53 97.444396,52.043513 97.109375,50.75 L96.914062,50 L68,50 L68,5 Z M100,6 C99.4477153,6 99,6.44771525 99,7 C99,7.55228475 99.4477153,8 100,8 C100.552285,8 101,7.55228475 101,7 C101,6.44771525 100.552285,6 100,6 Z M78,10 L78,12 L86,12 L86,10 L78,10 Z M89,10 C88.4477153,10 88,10.4477153 88,11 C88,11.5522847 88.4477153,12 89,12 C89.5522847,12 90,11.5522847 90,11 C90,10.4477153 89.5522847,10 89,10 Z M93,10 C92.4477153,10 92,10.4477153 92,11 C92,11.5522847 92.4477153,12 93,12 C93.5522847,12 94,11.5522847 94,11 C94,10.4477153 93.5522847,10 93,10 Z M100,10 C99.4477153,10 99,10.4477153 99,11 C99,11.5522847 99.4477153,12 100,12 C100.552285,12 101,11.5522847 101,11 C101,10.4477153 100.552285,10 100,10 Z M106,10 L106,12 L114,12 L114,10 L106,10 Z M117,10 C116.447715,10 116,10.4477153 116,11 C116,11.5522847 116.447715,12 117,12 C117.552285,12 118,11.5522847 118,11 C118,10.4477153 117.552285,10 117,10 Z M121,10 C120.447715,10 120,10.4477153 120,11 C120,11.5522847 120.447715,12 121,12 C121.552285,12 122,11.5522847 122,11 C122,10.4477153 121.552285,10 121,10 Z M100,14 C99.4477153,14 99,14.4477153 99,15 C99,15.5522847 99.4477153,16 100,16 C100.552285,16 101,15.5522847 101,15 C101,14.4477153 100.552285,14 100,14 Z M78,16 L78,18 L86,18 L86,16 L78,16 Z M89,16 C88.4477153,16 88,16.4477153 88,17 C88,17.5522847 88.4477153,18 89,18 C89.5522847,18 90,17.5522847 90,17 C90,16.4477153 89.5522847,16 89,16 Z M93,16 C92.4477153,16 92,16.4477153 92,17 C92,17.5522847 92.4477153,18 93,18 C93.5522847,18 94,17.5522847 94,17 C94,16.4477153 93.5522847,16 93,16 Z M106,16 L106,18 L114,18 L114,16 L106,16 Z M117,16 C116.447715,16 116,16.4477153 116,17 C116,17.5522847 116.447715,18 117,18 C117.552285,18 118,17.5522847 118,17 C118,16.4477153 117.552285,16 117,16 Z M121,16 C120.447715,16 120,16.4477153 120,17 C120,17.5522847 120.447715,18 121,18 C121.552285,18 122,17.5522847 122,17 C122,16.4477153 121.552285,16 121,16 Z M100,18 C99.4477153,18 99,18.4477153 99,19 C99,19.5522847 99.4477153,20 100,20 C100.552285,20 101,19.5522847 101,19 C101,18.4477153 100.552285,18 100,18 Z M78,22 L78,24 L86,24 L86,22 L78,22 Z M93,22 C92.4477153,22 92,22.4477153 92,23 C92,23.5522847 92.4477153,24 93,24 C93.5522847,24 94,23.5522847 94,23 C94,22.4477153 93.5522847,22 93,22 Z M100,22 C99.4477153,22 99,22.4477153 99,23 C99,23.5522847 99.4477153,24 100,24 C100.552285,24 101,23.5522847 101,23 C101,22.4477153 100.552285,22 100,22 Z M106,22 L106,24 L114,24 L114,22 L106,22 Z M121,22 C120.447715,22 120,22.4477153 120,23 C120,23.5522847 120.447715,24 121,24 C121.552285,24 122,23.5522847 122,23 C122,22.4477153 121.552285,22 121,22 Z M100,26 C99.4477153,26 99,26.4477153 99,27 C99,27.5522847 99.4477153,28 100,28 C100.552285,28 101,27.5522847 101,27 C101,26.4477153 100.552285,26 100,26 Z M78,28 L78,30 L86,30 L86,28 L78,28 Z M89,28 C88.4477153,28 88,28.4477153 88,29 C88,29.5522847 88.4477153,30 89,30 C89.5522847,30 90,29.5522847 90,29 C90,28.4477153 89.5522847,28 89,28 Z M93,28 C92.4477153,28 92,28.4477153 92,29 C92,29.5522847 92.4477153,30 93,30 C93.5522847,30 94,29.5522847 94,29 C94,28.4477153 93.5522847,28 93,28 Z M100,30 C99.4477153,30 99,30.4477153 99,31 C99,31.5522847 99.4477153,32 100,32 C100.552285,32 101,31.5522847 101,31 C101,30.4477153 100.552285,30 100,30 Z M78,34 L78,36 L86,36 L86,34 L78,34 Z M93,34 C92.4477153,34 92,34.4477153 92,35 C92,35.5522847 92.4477153,36 93,36 C93.5522847,36 94,35.5522847 94,35 C94,34.4477153 93.5522847,34 93,34 Z M100,34 C99.4477153,34 99,34.4477153 99,35 C99,35.5522847 99.4477153,36 100,36 C100.552285,36 101,35.5522847 101,35 C101,34.4477153 100.552285,34 100,34 Z M100,38 C99.4477153,38 99,38.4477153 99,39 C99,39.5522847 99.4477153,40 100,40 C100.552285,40 101,39.5522847 101,39 C101,38.4477153 100.552285,38 100,38 Z M100,42 C99.4477153,42 99,42.4477153 99,43 C99,43.5522847 99.4477153,44 100,44 C100.552285,44 101,43.5522847 101,43 C101,42.4477153 100.552285,42 100,42 Z" id="icon-book"></path>
                    </g>
                </g>
            </g>
        </g>
    </g>
  </SvgIcon>
));

export default BookIcon;
