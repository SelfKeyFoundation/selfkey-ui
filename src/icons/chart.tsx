import * as React from 'react';
import injectSheet, { StyleSheet, StyledComponentProps } from 'react-jss';

import { SvgIcon } from '@material-ui/core';
import { SvgIconProps } from '@material-ui/core/SvgIcon';

const styles: StyleSheet = {
  root: {
    width: '32px !important',
    height: '32px !important'
  }
};

export const ChartIcon = injectSheet(styles)<SvgIconProps>((props: SvgIconProps & StyledComponentProps)=> (
  <SvgIcon {...props} viewBox='0 0 32 32'>
    <g id="Corporate-Vault-chart" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
        <g id="802---Add-New-Member" transform="translate(-759.000000, -345.000000)" fill="#23E6FE" stroke="#23E6FE" strokeWidth="0.7">
            <g id="Modal" transform="translate(240.000000, 142.000000)">
                <g id="Select-role" transform="translate(185.000000, 127.000000)">
                    <g id="Radio-button-replace" transform="translate(0.000000, 52.000000)">
                        <g id="Btn-shareholder" transform="translate(310.000000, 0.000000)">
                            <path d="M40.7061106,25.2600751 C48.2005047,25.264512 54.2969099,31.3513607 54.2969099,38.8324439 C54.2969099,38.8945615 54.2955447,38.9897859 54.2941795,39.0710166 L40.9238638,39.0710166 L40.705428,38.9235725 L40.7061106,25.2600751 Z M54.457665,39.4123225 C54.6382158,39.4123225 54.6382158,39.3641984 54.6382158,38.8730592 C54.6382158,31.2008448 48.3848098,25 40.6986019,25 L40.5344338,25.0774764 L40.3641222,25.0774764 L40.3641222,39.1048059 L40.8190829,39.4123225 L54.457665,39.4123225 Z M38.880568,54.6586941 C31.4151849,54.6586941 25.3413059,48.573552 25.3413059,41.0938341 C25.3413059,33.7424472 31.2991409,27.6955313 38.6576953,27.5344349 L38.6576953,41.3289938 L38.8781788,41.7054542 L38.9990011,41.6648388 L38.9990011,41.7262738 L38.9303986,41.7682545 L49.5241912,49.5537824 C46.9647386,52.800625 43.0956953,54.6586941 38.880568,54.6586941 L38.880568,54.6586941 Z M38.9990011,41.2504935 L38.9990011,27.1890334 L38.8266417,27.1907399 C31.2025514,27.2617315 25,33.4984135 25,41.0938341 C25,48.7616115 31.2267841,55 38.880568,55 C43.26157,55 47.2787399,53.0405631 49.9020168,49.6240914 L50.0085042,49.4855212 L39.1293799,41.5180773 L38.9990011,41.2504935 Z M52.3931741,48.6706876 L42.1857398,41.1189542 L55.0768621,41.1189542 C54.9560398,43.7688529 53.9618158,46.5661957 52.3931741,48.6706876 L52.3931741,48.6706876 Z M41.1505591,40.7776483 L52.4607527,49.1451028 L52.5628031,49.0123348 C54.2669433,46.7928228 55.3369371,43.780116 55.4249941,40.9534209 L55.4301136,40.7776483 L41.1505591,40.7776483 Z" id="icon-chart-cv"></path>
                        </g>
                    </g>
                </g>
            </g>
        </g>
    </g>
  </SvgIcon>
));

export default ChartIcon;
