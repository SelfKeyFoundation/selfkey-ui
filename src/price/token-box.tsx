import * as React from 'react';
// @ts-ignore
import injectSheet, { StyleSheet, StyledComponentProps } from 'react-jss';

import { SvgIconProps } from '@material-ui/core/SvgIcon';
import { Grid } from '@material-ui/core'

import { TransferIcon } from '../icons/transfer';
import { Copy } from '../common/copy';

const styles: StyleSheet = {

    tokenBox: {
      minWidth: '320px',
      position: 'relative',
      padding: '20px 20px 8px 20px',
      borderRadius: '4px',
      backgroundColor: '#262f39',
      border: 'solid 1px #303c49',
      height: '186px'
    },

    tokenBoxBody: {
      marginTop: '20px'
    },

    tokenBoxFooter: {
      bottom: '7px',
      marginTop: '10px'
    },

    horizontalDivider: {
      height: '1px',
      backgroundColor: '#303c49'
    },

    addressBox: {
      marginTop: '10px',
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'center'
    },

    publicKey: {
      outline: 'none',
      cursor: 'pointer',
      fontFamily: 'Lato, arial, sans-serif',
      fontSize: '12px',
      color: '#93b0c1'
    },

    transferButton: {
      cursor: 'pointer',
      background: 'none',
      border: 'none',
      outline: 'none',
      display: 'flex',
      width: '100%',
      justifyContent: 'flex-end'
    },

    cryptoCurrencyName: {
      fontFamily: 'Lato, arial, sans-serif',
      fontSize: '18px',
      color: '#ffffff'
    },

    cryptoCurrencyShort: {
      marginTop: '4px',
      opacity: 0.7,
      fontFamily: 'Lato, arial, sans-serif',
      fontSize: '14px',
      letterSpacing: '1px',
      color: '#ffffff'
    }
};

export type TokenBoxProps = {
    cryptoCurrencyShort: string,
    cryptoCurrencyName: string,
    CryptoCurrencyIconComponent: React.ComponentType<SvgIconProps>,
    publicKey: string,
    transferAction?: ((event: React.MouseEvent<HTMLElement>) => void)
}

export const TokenBox = injectSheet(styles)<TokenBoxProps>(({classes, children, cryptoCurrencyShort, cryptoCurrencyName, CryptoCurrencyIconComponent, publicKey, transferAction }) => (
   <div className={classes.tokenBox}>
    <Grid container>
      <Grid item sm={2}>
        <CryptoCurrencyIconComponent/>
      </Grid>
      <Grid item sm={8}>
        <div className={classes.cryptoCurrencyName}>{cryptoCurrencyName}</div>
        <div className={classes.cryptoCurrencyShort} >{cryptoCurrencyShort}</div>
      </Grid>
      <Grid item sm={2}>
        <button className={classes.transferButton} onClick={transferAction}>
          <TransferIcon/>
        </button>
      </Grid>
    </Grid>
    <Grid className={classes.tokenBoxBody}>
      {children}
    </Grid>
    <Grid className={classes.tokenBoxFooter}>
        <div className={classes.horizontalDivider}></div>
    </Grid>
    <Grid container className={classes.addressBox}>
      <Grid item xs={10}>
        <span className={classes.publicKey}>
          {publicKey}
        </span>
      </Grid>
      <Grid item xs={2}>

        <Copy text={publicKey}>
        </Copy>

      </Grid>
    </Grid>
  </div>
));

export default TokenBox;
