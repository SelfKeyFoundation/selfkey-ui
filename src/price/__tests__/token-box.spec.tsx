import * as React from 'react';
import { create } from 'react-test-renderer';
import { TokenBox } from '../token-box';
import { SelfkeyIcon } from '../../icons/selfkey';


describe('Token Box', () => {
  it('should show Token Box for key', () => {
    const component = create(
      <TokenBox  cryptoCurrencyShort='KEY' cryptoCurrencyName='Selfkey' CryptoCurrencyIconComponent={SelfkeyIcon} publicKey='0x4184288c556524df9cb9e58b73265ee66dca4efe'/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});





