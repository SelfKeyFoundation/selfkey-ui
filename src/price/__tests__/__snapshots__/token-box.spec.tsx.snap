// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Token Box should show Token Box for key 1`] = `
<div
  className="Component-tokenBox-0-1-1"
>
  <div
    className="MuiGrid-container-1"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-sm-2-45"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-98 Component-root-0-1-10"
        focusable="false"
        role="presentation"
        viewBox="0 0 44 44"
      >
        <g
          fill="none"
          fillRule="evenodd"
        >
          <rect
            fill="#2DA1F8"
            height="44"
            rx="9"
            width="44"
          />
          <path
            d="M33.2899445,22.9203128 C31.3240959,20.9415891 28.2827759,20.6709767 26.0176127,22.1008593 L14.3790117,10.384719 C13.8693908,9.87176033 13.0437885,9.87176033 12.5342002,10.384719 L8.38221571,14.5638959 C7.87259476,15.0768546 7.87259476,15.9078641 8.38221571,16.4207899 C8.89183732,16.9337157 9.71743957,16.9337486 10.2270279,16.4207899 L13.4554644,13.171209 L15.7614951,15.4923429 L12.5330586,18.7419238 C12.0234377,19.2548825 12.0234377,20.0858919 12.5330586,20.5988178 C13.0426796,21.1117436 13.8682818,21.1117765 14.3778702,20.5988178 L17.6063067,17.3492369 L24.1715292,23.9574578 C22.7509506,26.237555 23.0210741,29.2987939 24.9856507,31.2774191 C27.2674805,33.5741936 31.0068426,33.5741936 33.2886398,31.2774191 C35.5704371,28.9806446 35.5704697,25.2167919 33.2886398,22.9200502 L33.2899445,22.9203128 Z M31.4451329,24.7772068 C32.7293777,26.0698627 32.7293777,28.1267859 31.4451329,29.4196059 C30.1608881,30.7124259 28.1173489,30.7122618 26.832941,29.4196059 C25.5485331,28.12695 25.5486962,26.0700268 26.832941,24.7772068 C28.1171858,23.4843868 30.1607251,23.4845509 31.4451329,24.7772068 Z"
            fill="#FFFFFF"
            id="Page-1-Copy"
            transform="translate(21.500000, 21.500000) scale(-1, 1) rotate(45.000000) translate(-21.500000, -21.500000) "
          />
        </g>
      </svg>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-sm-8-51"
    >
      <div
        className="Component-cryptoCurrencyName-0-1-8"
      >
        Selfkey
      </div>
      <div
        className="Component-cryptoCurrencyShort-0-1-9"
      >
        KEY
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-sm-2-45"
    >
      <button
        className="Component-transferButton-0-1-7"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-98 Component-root-0-1-11"
          focusable="false"
          role="presentation"
          viewBox="0 0 489.2 489.2"
        >
          <g>
            <g>
              <path
                d="M481.044,382.5c0-6.8-5.5-12.3-12.3-12.3h-418.7l73.6-73.6c4.8-4.8,4.8-12.5,0-17.3c-4.8-4.8-12.5-4.8-17.3,0l-94.5,94.5    c-4.8,4.8-4.8,12.5,0,17.3l94.5,94.5c2.4,2.4,5.5,3.6,8.7,3.6s6.3-1.2,8.7-3.6c4.8-4.8,4.8-12.5,0-17.3l-73.6-73.6h418.8    C475.544,394.7,481.044,389.3,481.044,382.5z"
              />
              <path
                d="M477.444,98l-94.5-94.4c-4.8-4.8-12.5-4.8-17.3,0s-4.8,12.5,0,17.3l73.6,73.6h-418.8c-6.8,0-12.3,5.5-12.3,12.3    s5.5,12.3,12.3,12.3h418.8l-73.6,73.4c-4.8,4.8-4.8,12.5,0,17.3c2.4,2.4,5.5,3.6,8.7,3.6s6.3-1.2,8.7-3.6l94.5-94.5    C482.244,110.6,482.244,102.8,477.444,98z"
              />
            </g>
          </g>
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
          <g />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="Component-tokenBoxBody-0-1-2"
  />
  <div
    className="Component-tokenBoxFooter-0-1-3"
  >
    <div
      className="Component-horizontalDivider-0-1-4"
    />
  </div>
  <div
    className="MuiGrid-container-1 Component-addressBox-0-1-5"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-10-39"
    >
      <span
        className="Component-publicKey-0-1-6"
      >
        0x4184288c556524df9cb9e58b73265ee66dca4efe
      </span>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-2-31"
    >
      <button
        className="CopyComponent-copyButton-0-1-12"
        onClick={[Function]}
      >
        <span
          className="CopyComponent-copyButtonSpan-0-1-13"
        >
          COPY
        </span>
      </button>
    </div>
  </div>
</div>
`;
